pkg/providers/oidc/verifier.go

```diff
@@ -8,18 +8,28 @@ import (
 	"github.com/coreos/go-oidc/v3/oidc"
 )
 
-// idTokenVerifier allows an ID Token to be verified against the issue and provided keys.
+// IDTokenVerifier allows an ID Token to be verified against the issue and provided keys.
 type IDTokenVerifier interface {
 	Verify(context.Context, string) (*oidc.IDToken, error)
 }
 
+// IDTokenExpirationCheck allows an ID Token to be verified if it is expired.
+type IDTokenExpirationCheck interface {
+	Verify(context.Context, string) (bool, error)
```

Comment:
Using the same name here means you couldn't implement a verifier that verifies the token and the expiration at the same time. Wondering if we should call this something different?